### DISCUSSION TITLE
Using dectalk after building

### DISCUSSION THREAD

**Discussion by b7YPWQvNLBrMQW** [dectalk/dectalk#40] [Q&A] [answered]
Hello, sorry if this is an extremely basic question, but I don't build from source very often.
I can't figure out how to actually use dectalk (either `say` or `speak`) after the build process.
I went with the Docker build option, running on Ubuntu 22.04.2 LTS. I believe the docker-compose finished successfully, the image exists and the container exited gracefully. Full container logs - [dectalk-json.log](https://github.com/dectalk/dectalk/files/12195517/dectalk-json.log)
From here, I intend to mostly use dectalk via CLI, but I'm not sure what the next step is. I can see a `makefile` in `/src/samples/SAY/` that looks like it will create `say.exe`, but I thought had seen references that `/src/samplosf/build` was the proper directory for Linux systems.
Hopefully I'm just overthinking it, but I appreciate any advice or instructions.
Thanks

**Comment by stargo** [comment, marked as answer]
After the docker-compose build-process you should now have a directory called "dist" at the top-level which contains all the built binaries. You should be able to start say in there `./say` or `./dist/say` from the top-level.

If you want to look at the source, `src/samplosf` is what is used on unix, `src/samples` are the windows examples.

**Reply by scruss** [reply]
For the regular build,

    sudo make install

will install everything under /opt/dectalk, with binaries symlinked to /usr/bin (or wherever you told `./configure` to put them):

```
$ which say
/usr/bin/say
$ ls -la /usr/bin/say
lrwxrwxrwx 1 root root 16 Jul 29 15:06 /usr/bin/say -> /opt/dectalk/say
```
